### PR TITLE
Exp/Log layers

### PR DESCRIPTION
**Pull request by hansely**
Support for exp/log layer and shape
Added model compiler support for upsample layer

### CODE REVIEW

**Comment by hansely** [comment]
Not ready yet

File: amd_openvx_extensions/amd_nn/src/tensor_log.cpp [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+                "#pragma OPENCL EXTENSION cl_khr_fp16 : enable\n"
+                "__kernel void %s(__global uchar * in, uint in_offset, uint4 in_stride, __global uchar * out, uint out_offset, uint4 out_stride) \n"
+                "{ \n"
+                "     uint x = get_global_id(0);\n"
+                "     uint y = get_global_id(1);\n"
+                "     uint c = get_global_id(2);\n"
+                "     half value = *(__global half *)&in[in_offset + x * in_stride.s0 + y * in_stride.s1 + c * in_stride.s2];\n"
+                "     out += out_offset + x  * out_stride.s0 + y * out_stride.s1 + c * out_stride.s2;\n"
+                "     // read 1 value and calculate its natural log\n"
+                "     *(__global half *)&out[0] = log(value);\n"
```
Comment (rrawther) [inline amd_openvx_extensions/amd_nn/src/tensor_log.cpp]: probably need to use half_log()

Comment (hansely) [reply amd_openvx_extensions/amd_nn/src/tensor_log.cpp]: Will update the code. Thanks!

**Comment by hansely** [comment]
Ready to be merged

**Comment by hansely** [comment]
-Fixed the convolution bug
-Passed the profiler
-Ready to be merged

**Pull request merged**